### PR TITLE
Update build workflow and remove unused scripts

### PR DESCRIPTION
**Pull request by kraenhansen**
This updates the build workflow with a couple of fixes to issues discovered while publishing and removes scripts that are no longer used as we've transitioned to GitHub actions.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,14 +34,14 @@ jobs:
...
         with:
-          node-version: 20
+          node-version: 18
       - name: Install wine
         run: |
           sudo dpkg --add-architecture i386
           sudo apt-get update
           sudo apt install -y wine wine32
       - name: Install dependencies
-        run: npm ci --legacy-peer-deps
```
Comment (kraenhansen) [inline .github/workflows/build.yml:38]: Seems this isn't necessary after all.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -52,7 +58,12 @@ jobs:
       # Building for additional architectures works without conflicts - so to
       # enable arm64 builds for Windows and Linux just add --arm64 here
       - name: Prepare installer contents
-        run: npx electron-builder -lw --publish never --dir --x64
+        run: |
+          npx electron-builder -lw --publish never --dir --x64
+          # Copying over the app-update.yml from Linux as it is skipped when using --dir
+          # https://github.com/electron-userland/electron-builder/blob/79df54238621fbe48ba20444129950ba2dc49983/packages/app-builder-lib/src/publish/PublishManager.ts#L113-L115
+          cat dist/linux-unpacked/resources/app-update.yml
```
Comment (kneth) [inline .github/workflows/build.yml:65]: Is it needed?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -28,14 +34,14 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v3
         with:
-          node-version: 20
+          node-version: 18
```
Comment (kraenhansen) [inline .github/workflows/build.yml:37]: Aligning with the version used by the Electron dependency.

**Review by kneth — APPROVED** [review]
Bye, bye, Jenkins

File: .github/workflows/build.yml (hunk 2 above)
Comment (kraenhansen) [reply .github/workflows/build.yml:65]: Not strictly needed, but I wanted to be sure it exists (as it's relying on undocumented behaviour of electron-builder) and displaying its contents might help debugging future issues.

**Pull request merged**